Add support for Google Analytics for WooCommerce version 2.0.0

martynmjones opened the pull request:
### Changes proposed in this Pull Request:

⚠️ **Requires:** https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/322

Google Analytics for WooCommerce updates how the Global Site Tag is loaded so this PR updates the integration in GLA to support version `2.0.0` and above while maintaining backwards compatibility.

The main difference is that instead of simply adding a script to the page with a call to `gtag`, we now use `wp_add_inline_script` and the Google Analytics for WooCommerce script handle where applicable.

### Detailed test instructions:

1. Build Google Analytics for WooCommerce from [#322](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/322)
2. Complete onboarding if needed
3. Debug a GLA test site via [TagAssistant](https://tagassistant.google.com/)
4. Confirm the GLA tag is found
5. Confirm that the GLA events are sent:
    - `page_view`
    - `view_item`
    - `purchase`
    - `conversion`

### Changelog entry

> Add - Support for Google Analytics for WooCommerce version 2.0.0 and above

martynmjones committed:
Add support for Google Analytics for WooCommerce version 2.0.0

martynmjones committed:
phpcs

tomalec commented on src/Google/GlobalSiteTag.php:
```diff
@@ -217,16 +217,24 @@ function () use ( $gtag_events ) {
-					);
-				}
-			);
+			if ( version_compare( \WC_GOOGLE_ANALYTICS_INTEGRATION_VERSION, '2.0.0', '>=' ) ) {
```
:construction: 
I don't get into this line when I run it with the new GA, as the one above returns `false`.
GLA still expect the old GA settings to be there https://github.com/woocommerce/google-listings-and-ads/blob/3a423a2d5cfca64486c74cf41905d01db1cc4c52/src/Proxies/GoogleGtagJs.php#L24:L33

The latest GA does not set `ga_gtag_enabled`

tomalec replied:
I've made a fix in https://github.com/woocommerce/google-listings-and-ads/compare/update/global-site-tag-for-analytics-version-2...update/global-site-tag-for-analytics-version-2-enabled?expand=1

tomalec committed:
Consider `ga_gtag_enabled=yes` for  WCGAI >= 2

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/2287#discussion_r1508145827

martynmjones committed:
Merge pull request #2288 from woocommerce/update/global-site-tag-for-analytics-version-2-enabled

Consider `ga_gtag_enabled=yes` for  WCGAI >= 2

tomalec commented on src/Google/GlobalSiteTag.php:
```diff
@@ -249,16 +257,44 @@ function gtag() { dataLayer.push(arguments); }
-				'groups': 'GLA',
-				'send_page_view': false
-			});
+			<?php echo esc_js( $this->get_gtag_config( $ads_conversion_id ) ); ?>
```
:construction: 
Escaping JS, results in syntax errors for escaped quotes
```suggestion
			<?php echo $this->get_gtag_config( $ads_conversion_id ); ?>
```

martynmjones committed:
Remove esc_js from gtag config

Co-authored-by: Tomek Wytrębowicz <tomalecpub@gmail.com>

martynmjones committed:
Ignore OutputNotEscaped warning for gtag config output

tomalec reviewed (approved):
Reviewed the code, tested locally LGTM, thanks! :)
